src/ui/public/agg_response/_embedded_tooltip_formatter.js

```diff
@@ -0,0 +1,122 @@
+import _ from 'lodash';
+import $ from 'jquery';
+import { VisRequestHandlersRegistryProvider } from 'ui/registry/vis_request_handlers';
+import { VisResponseHandlersRegistryProvider } from 'ui/registry/vis_response_handlers';
+import { FilterBarQueryFilterProvider } from 'ui/filter_bar/query_filter';
+
+export function EmbeddedTooltipFormatterProvider($rootScope, $compile, Private, getAppState, savedVisualizations) {
+  const queryFilter = Private(FilterBarQueryFilterProvider);
```

Comment:
whats the reason you are handling requests/responses on your own ? i am wondering if it would be possible to use <visualize> and let it take care of all this ?
if i understand correctly you just load a saved visualization + add additional filter to it ?
that should be possible using `<visualize saved-id='...' app-state='appState'>` where appState should have a property filters

Reply:
I need to update the saved search with the context of the moused-over buckets and then fetch the visualization for that context. If there is a another way then please let me know.

Reply:
i think something like this could work:

```
const visualizationId = '';
const visContainer = $('#the-container-element');
$scope.timeRange = { from: ..., to: ... };
savedVisualizations.get(visualizationId).then(savedObj => {
      $scope.savedObj = savedObj;
      const filter = aggResult.aggConfig.createFilter(aggResult.key) // aggResult of the bucket you clicked
      $scope.savedObj.searchSource.get('filter').push(filter);
      const visEl = $('<visualize saved-obj="savedObj" time-range="timeRange"></visualize>');
      const visHtml = $compile(visEl)($scope);
      visContainer.html(visHtml);
});
```

Reply:
If I go that route, how will I know when the response has been fetched? I need to be able to show a loading page until the visualization has data. When the visualization gets data, then I show it.